### PR TITLE
Wait storage operations instead of shutdown on unhandled exception

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/ChannelGroup.cs [hunk 1]
```diff
@@ -9,19 +9,30 @@ namespace Microsoft.AppCenter.Channel
 {
     public sealed class ChannelGroup : IChannelGroup, IAppSecretHolder
     {
+        private static readonly TimeSpan WaitStorageTimeout = TimeSpan.FromSeconds(5);
```
Comment (guperrot) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/ChannelGroup.cs]: Shouldn't it be _waitStorageTimeout?

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/IStorage.cs [hunk 2]
```diff
@@ -58,6 +58,13 @@ public interface IStorage : IDisposable
         /// <exception cref="StorageException"/>
         Task<string> GetLogsAsync(string channelName, int limit, List<Log> logs);
 
+        /// <summary>
+        /// Waits for any running storage operations to complete.
+        /// </summary>
+        /// <param name="timeout">The maximum amount of time to wait for remaining tasks</param>
+        /// <returns>True if remaining tasks completed in time; false otherwise</returns>
+        Task<bool> WaitAsync(TimeSpan timeout);
```
Comment (guperrot) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/IStorage.cs]: Either add Operations to method name or remove Operations to the Channel method to make both consistent.

Comment (guperrot) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/IStorage.cs]: Is boolean needed? I don't think so.

File: Tests/Microsoft.AppCenter.Test.UWP/AppCenterTest.cs [hunk 3]
```diff
@@ -103,11 +102,34 @@ public void StartServiceAfterUnhandledException()
...
             // Some exception occurred.
             ApplicationLifecycleHelper.Instance.InvokeUnhandledExceptionOccurred(null, new Exception());
+            Task.Delay(100).Wait();
 
             // Start any service.
             AppCenter.Start(typeof(TestAppCenterService));
 
-            // No exception throws, just log message.
+            // No exceptions are thrown.
+            // TODO: Verify that there are no errors in logs (System.Diagnostics.Debug.Listeners is not available here).
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.UWP/AppCenterTest.cs]: I think we will not address that todo, please remove. Maybe keep comment that we cannot verify this but no TODO keyword.

File: Tests/Microsoft.AppCenter.Test.UWP/AppCenterTest.cs [hunk 4]
```diff
@@ -103,11 +102,34 @@ public void StartServiceAfterUnhandledException()
...
+            // Some exception occurred.
+            ApplicationLifecycleHelper.Instance.InvokeUnhandledExceptionOccurred(null, new Exception());
+            Task.Delay(100).Wait();
+
+            // Set custom properties.
+            AppCenter.SetCustomProperties(new CustomProperties().Set("test", 42));
+            Task.Delay(100).Wait();
+
+            // No exceptions are thrown.
+            // TODO: Verify that there are no errors in logs (System.Diagnostics.Debug.Listeners is not available here).
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.UWP/AppCenterTest.cs]: Same comment as for the similar todo.

**Comment by guperrot** [comment]
(This is the PR to address #613.)

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/ChannelGroup.cs (hunk 1 above)
Comment (MatkovIvan) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/ChannelGroup.cs]: Nope I guess, this is constant, not class field.

**Pull request merged**